tsl/src/bgw_policy/job.c

```diff
@@ -89,14 +89,65 @@ get_chunk_id_to_reorder(int32 job_id, Hypertable *ht)
 															 -1);
 }
 
+static int64
+get_compression_window_end_value(Dimension *dim, Jsonb *config)
+{
+	Oid partitioning_type = ts_dimension_get_partition_type(dim);
+
+	if (IS_INTEGER_TYPE(partitioning_type))
+	{
+		int64 lag = policy_compression_get_older_than_int(config);
+		Oid now_func = ts_get_integer_now_func(dim);
+		ts_interval_now_func_validate(now_func, partitioning_type);
+
+		if (InvalidOid == now_func)
+			ereport(ERROR,
+					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					 errmsg("integer_now function must be set")));
```

Comment:
It's already validated in `ts_interval_now_func_validate`.

Reply:
there is some intermediate code here that i want to fully clean up after migrating drop_chunk because that allows me to remove ts_interval and have some shared functions between drop chunk and compression policy

Reply:
Can you move this check before calling `ts_interval_now_func_validate`? Then it will be reasonable, since it produces new error message.

Reply:
removed the call, since it is indeed redundant as ts_get_integer_now_func already does the check